src/main.rs

```diff
@@ -69,16 +76,18 @@ fn main() {
     // Parse & Build an AST
     let mut parser = Parser::new(lexer);
 
-    let ast_root = match parser.parse() {
+    let mut ast_root = match parser.parse() {
```

Comment:
Semantic Analysis needs to be able to modify the tree. IE the type checker inferring the type.